src/future-proofing.md

```diff
@@ -202,3 +202,21 @@ on the data structure.
 [`std::borrow::Cow`]: https://doc.rust-lang.org/std/borrow/enum.Cow.html
 [`std::boxed::Box`]: https://doc.rust-lang.org/std/boxed/struct.Box.html
 [`std::io::BufWriter`]: https://doc.rust-lang.org/std/io/struct.BufWriter.html
+
+<a id="c-nightly-optin"></a>
+## Nightly features use an explicit opt-in (C-NIGHTLY-OPTIN)
+
+Some libraries need to use, or want to experiment with, the [nightly channel].
+To avoid accidental breakage, libraries should either:
+- Use nightly features unconditionally, so that people depending on the library must always use a nightly toolchain to build
+- Add a `--cfg` flag which opts-in to the nightly features (optionally, with feature detection to verify the features are present in the current compiler version). This allows people to avoid opting-in if they do not want to be exposed to possible breakage.
```

Comment:
```suggestion
- Add a `--cfg` flag which opts-in to the nightly features (optionally, with a test compilation to provide better error messages from opt-in when feature is not supported). This allows people to avoid opting-in if they do not want to be exposed to possible breakage.
```

Reply:
Perhaps something like this? It's a stronger statement and less likely to be misread in a way that suggests avoiding the explicit opt-in.

Reply:
sure, i added something along those lines.